session.go

```diff
@@ -2290,6 +2291,51 @@ func (t *traceWriter) Trace(traceId []byte) {
 	t.mu.Lock()
 	defer t.mu.Unlock()
 
+	fetchAttempts := 1
+	if t.maxAttempts > 0 {
+		fetchAttempts = t.maxAttempts
+	}
+
+	for i := 0; i < fetchAttempts; i++ {
+		isDone := false
+		iter := t.session.control.query(`SELECT event_id, activity, source, source_elapsed, thread
+			FROM system_traces.events
+			WHERE session_id = ?`, traceId)
+
+		for iter.Scan(&timestamp, &activity, &source, &elapsed, &thread) {
+			if strings.Contains(activity, "Done processing") {
+				isDone = true
+				break
+			}
+		}
+
+		if err := iter.Close(); err != nil {
+			fmt.Fprintln(t.w, "Error:", err)
+			return
+		}
+
+		if isDone || i == fetchAttempts-1 {
+			break
+		}
+
+		time.Sleep(t.sleepInterval)
+	}
+
```

Comment:
Let's keep it your way, only thing please write a message to the buffer when trace is incomplete.